tests/e2e/test_rosacli_version.go

```diff
@@ -0,0 +1,52 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/rosa/pkg/info"
+	"github.com/openshift/rosa/tests/ci/labels"
+	"github.com/openshift/rosa/tests/utils/exec/rosacli"
+)
+
+var _ = Describe("Get CLI version",
+	labels.Day1,
```

Comment:
I would have put that one as Day2. We don't have any created resource here

Reply:
We do `rosa login` beyond the test cases. It is complex for us to call `rosa login` back after logout. logout is a destructive operation.

I think validating `rosa version` is more meaningful rather than `logout -> rosa version`.

Reply:
I'm fine with changing the label(s) if needed, but the bug this is related to is specifically about testing that a user can still get the version number without logging in. If logging out is too destructive I could change this to just checking the version and mark the polarion case as manual only if necesary

Reply:
isn't there an other way to set the OCM config env in the command run to a temp file and never login. That would reproduce the "not logged in" behavior to run the `rosa version` command

Reply:
We could run such `logout` cases in an individual step/job. But they need a label(maybe `destructive-logout`) rather than `day1`, `day2`, and so on. These cases are so special, even in destructive scenarios.

Reply:
@jerichokeyne Could you mark the label `destructive-logout` on this case?

Reply:
@yasun1 I added the label `destructive-logout` to this specific case, and I also changed the overall label to `day2` in case we want to add more version tests

Reply:
@yasun1 @jerichokeyne Why not applying the `OCM_CONFIG` env as suggested [here](https://github.com/openshift/rosa/pull/2051#discussion_r1603855548) ?
This would leave the cluster and OCM connection intact and we would not need an extra destructive label on this case
cc @xueli181114

Reply:
@radtriste 
> @yasun1 @jerichokeyne Why not applying the `OCM_CONFIG` env as suggested [here](https://github.com/openshift/rosa/pull/2051#discussion_r1603855548) instead of doing a logout ? This would leave the cluster and OCM connection intact and we would not need an extra destructive label on this case cc @xueli181114

I mocked up a quick way of doing that. I couldn't find an easy way to pass environment variables to the rosa runner, so I just set the environment variable outside of that. But, I'm not sure if that could cause issues with running other tests in parallel

Reply:
@jerichokeyne gonna have a look

Reply:
@jerichokeyne To setup an env to the client, you could:
1. Add a new field to the `runner` struct (like `var envs []string`)
2. Add a new method to the `runner` interface, `AddEnvVar(key,value)`,  which would add the `key=value` to the `envs` field
3. Add a new Method to the `Client` struct, `AddEnvVar(key,value)`, which would call the runner `AddEnvVar` method
4. In the `runner.Run()` method, add this line when setting up the command (line #183): `cmd.Env = append(cmd.Env, r.envs...)`
5. After you create your client in the `BeforeEach`, call the `rosaclient.AddEnvVar("OCM_CONFIG", tmpFile.Name())`

Also, the creation of the temp file for ocm config could be moved to `utils/common` as a `CreateTempOCMConfig` method, wdyt ?

Reply:
Yes, configuring the running environment could solve the issue. `rosa login` generates the credentials under `$HOME/.config/ocm/ocm.json`, which each rosa command will use to connect to the API. If the HOME path is changed, it is in `not login` status.

Reply:
@radtriste, I implemented those changes. Feel free to take another look